api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3301,13 +3301,11 @@ public void makeBucket(String bucketName, String region, boolean objectLock)
       throw new RegionConflictException(
           "passed region conflicts with the one previously specified");
     }
-    String configString;
-    if (region == null || US_EAST_1.equals(region)) {
-      // for 'us-east-1', location constraint is not required.  for more info
-      // http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
-      region = US_EAST_1;
-      configString = "";
-    } else {
+
+    String configString = "";
+    // Location constraint is not required for 'us-east-1' region. Refer
+    // http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
+    if (region != null && !region.equals(US_EAST_1)) {
       CreateBucketConfiguration config = new CreateBucketConfiguration(region);
       configString = config.toString();
```

Comment:
@balamurugana I think the issue is `config.toString()` which is not returning XML here

Reply:
Yep. Fixed.